doto-frontend/src/routes/Route.js

```diff
@@ -5,11 +5,39 @@ import Login from "../components/pages/Login/Login";
 import Calendar from "../components/pages/Calendar/Calendar";
 import NotFound from "../components/pages/NotFound";
 import { ThemeContext } from "../context/ThemeContext";
+import CookieManager from "../helpers/CookieManager";
 import "../tailwind-generated.css";
 
+/**
+ * When the user is redirected after they are logged in to their google account
+ * We take the current url and extract the Email and JWT Token from the query parameters
+ *
+ * There was also a weird issue where the JWT token with emails ending with "gmail.com" had a
+ * "#" character at the end but the emails ending with "aucklanduni.ac.nz" did not have this issue
+ * and that is why we are doing a hacky check to see if the email domain starts with "a".
+ */
```

Comment:
Would this issue persist with emails with other domains eg 'mit.ac.nz' in which case our method may fail?

Reply:
We are assuming that the users will be currently using a '@gmail.com and UoA email. If there are others, the backend might have to change.